atc/db/build.go

```diff
@@ -1244,6 +1244,16 @@ func (b *build) SaveOutput(
 		return err
 	}
 
+	// resource.SetResourceConfigScope has to be called after the transaction,
+	// because if resourceConfigScope is newly created by findOrCreateResourceConfigScope,
+	// before it's committed, SetResourceConfigScope will violate foreign key constraint.
+	if newVersion {
+		err = resource.SetResourceConfigScope(resourceConfigScope)
```

Comment:
Can we just do this within the same transaction instead? Otherwise there's potentially a race condition wherein GC removes the scope because it's not (yet) referenced by a resource, which would lead to this failing.

I think this method call really just does one pretty simple query so it should be OK to just inline it so it can use `tx`.

Reply:
Ah, I didn't read code of `resource.SetResourceConfigScope` when I made the initial change. Now I realize that, it always embeds a transaction, so that in my previous test, there were two different transactions.

Reply:
Fixed.